src/views/ApartmentApp/index.js

```diff
@@ -18,54 +19,123 @@ export default class ApartApp extends Component {
       isFac: Boolean,
       isAlu: Boolean,
       loading: true,
+      saving: false,
+      savingSuccess: false,
       network: 'online',
       submitDialogOpen: false, // Use this for saving app (later feature)
       errorDialogOpen: false,
-      errorDialogText: null,
       userProfile: {},
       applicants: [],
       // TODO - For end-to-end Hello World debug. Remove the next 2 lines before merge
       onCampusRoom: null,
       onOffCampus: null,
     };
+    this.errorDialogText = '';
+    this.snackbarText = '';
+    this.snackbarSeverity = '';
+    this.saveButtonAlertTimeout = null;
+  }
+
+  componentDidMount() {
+    this.loadProfile();
+    this.loadHousingInfo();
+    // this.checkForSavedApplication();
+  }
+
+  /**
+   * Loads the user's profile info only once (at start)
+   */
+  async loadProfile() {
+    this.setState({ loading: true });
+    try {
+      const profile = await user.getProfileInfo();
+      this.setState({ userProfile: profile });
+      this.setState({ isStu: String(profile.PersonType).includes('stu') });
+      if (this.state.isStu) {
+        let applicants = this.state.applicants;
+        applicants.push(profile);
+        this.setState({ applicants });
+      }
+      this.setState({ loading: false });
+    } catch (error) {
+      // Do Nothing
+    }
+  }
+
+  /**
+   * Loads the user's saved apartment application, if one exists
+   */
+  async loadHousingInfo() {
+    this.setState({ loading: true });
+    try {
+      // TODO - Once saving application has been implemented in the backend, this will be replaced with a call to the load the application info. The getHousingInfo was made obsolete after the Hello World
+      let housingInfo = await housing.getHousingInfo();
+      let onOffCampus = String(housingInfo[0].OnOffCampus);
+      let onCampusRoom = String(housingInfo[0].OnCampusRoom);
+      this.setState({ onOffCampus, onCampusRoom, loading: false });
+    } catch (error) {
+      // Do Nothing
+    }
   }
 
   /**
    * Callback for apartment people search submission
    * @param {String} searchSelection Username for student
    */
-  onSearchSubmit = (searchSelection) => {
+  onSearchSubmit = searchSelection => {
+    this.setState({ updating: true });
     if (searchSelection && searchSelection !== null) {
```

Comment:
Line 87, it  looks like you missed one of the redundant null checks.

Reply:
Oh, how did that one sneak passed 😆